aether-kernel/aether/kernel/api/avro_tools.py

```diff
@@ -468,7 +469,8 @@ def avro_schema_to_passthrough_artefacts(item_id, avro_schema):
         # this is an auto-generated mapping that shouldn't be modified manually
         'is_read_only': True,
         'is_active': True,
-        'input': definition,  # include avro schema
+        'input': sample,  # include a data sample
+        'schema': definition,  # include avro schema
```

Comment:
This should be the original AVRO schema, `definition` is the tweaked one with the missing id :grimacing:

```python
'input': random_avro(avro_schema),  # include a data sample
'schema': avro_schema,  # include avro schema
```

But not sure if I prefer to return the mappingset instance rather than include the fields withing the mapping definition :thinking:

Reply:
I mean this

```python
def upsert_project_with_avro_schemas(
    action='upsert',
    project_id=None,
    project_name=None,
    avro_schemas=[],
):
    '''
    Creates or updates the project and links it with the given AVRO schemas
    and related artefacts: project schemas and passthrough mappings.

    Returns the list of project and its affected artefact ids by type.
    '''

    schemas = []
    mappingsets = []
    mappings = []

    for avro_schema in avro_schemas:
        schema, mapping = parser(avro_schema.get('id'), avro_schema.get('definition'))
        schemas.append(schema)
        mappings.append(mapping)

        id = schema.get('id')
        mappingsets.append({
            'id': id,
            'name': schema.get('name'),
            'schema': avro_schema.get('definition'),
            'input': random_avro(avro_schema.get('definition')),
        })

        # Include an empty mapping if the schema is new
        if not Schema.objects.filter(pk=id).exists():
            mappings.append({
                # even being the same name it's going to append the suffix `-1`
                'name': schema.get('name'),
                # the passthrough mapping has created a mappingset with this id
                'mappingset': id,
            })

    return upsert_project_artefacts(
        action=action,
        project_id=project_id,
        project_name=project_name,
        schemas=schemas,
        mappingsets=mappingsets,
        mappings=mappings,
    )
```

Reply:
Understood, the cool thing is `upsert_project_artefacts` knows how to handle both

Reply:
:stuck_out_tongue: